pkg/bootstrap/option/convert.go

```diff
@@ -94,6 +95,9 @@ func tlsContextConvert(tls *networkingAPI.ClientTLSSettings, sniName string, met
 		CommonTlsContext: &auth.CommonTlsContext{},
 	}
 
+	// We always set v3, since we know this is a new proxy that supports v3
+	requestedType := v3.ListenerType
```

Comment:
nit: v3.ClusterType may be more appropriate as this is applied for cluster, but does not really matter